server/src/main/java/io/crate/analyze/WhereClause.java

```diff
@@ -81,7 +81,7 @@ public WhereClause(@Nullable Symbol normalizedQuery,
     }
 
     public boolean hasQuery() {
-        return query != null;
+        return query != null && !query.symbolType().isValueSymbol();
```

Comment:
Looking at some of the usages I think this could also lead to bugs if the query is a literal `false`. 

Maybe we should combine this with `canMatch` ?

Reply:
hm do you mean combine it with the method body logic `query != null && !query.symbolType().isValueSymbol()` or at the `hasQuery` call sides? Combining it in the method body probably would probably semantically incorrect? Another possible solution would be just using `canMatch` in the `DeletePlanner` instead of `!hasQuery` https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/planner/statement/DeletePlanner.java#L136 
and leave the `hasQuery` without further modifications.

Reply:
> Another possible solution would be just using canMatch in the DeletePlanner instead of !hasQuery

I like that solution :+1:

Reply:
Although I think it would also change semantics? If there still is a query, then it's not guranteed that it matches the entire query?

I think we'd have to be explicit and check that either there is no query, or it matches everything